R/tabyl.R

```diff
@@ -207,7 +207,12 @@ tabyl_2way <- function(dat, var1, var2, show_na = TRUE, show_missing_levels = TR
   tabl[2][is.na(tabl[2])] <- "NA_"
   tabl[2][tabl[2] == ""] <- "emptystring_"
   result <- tabl %>%
-    tidyr::spread(!!var2, "tabyl_2way_n", fill = 0)
+    tidyr::pivot_wider(
+      names_from = !!var2,
+      values_from =  "tabyl_2way_n",
+      values_fn = ~ dplyr::coalesce(.x, 0L),
+      names_sort = TRUE
+    )
```

Comment:
From `tidyr::spread()` documentation, fill = 0 had to be replaced with  dplyr::coalesce(.x, 0L) to make sure implict and explicit missing values were replaced.
Also, `spread()` sorted names automatically, not pivot_wider(), hence the names_sort argument.

```
@param fill If set, missing values will be replaced with this value. Note that there are two types of missingness in the input: explicit missing values (i.e. NA), and implicit missings, rows that simply aren't present. Both types of missing value will be replaced by fill.
```

Reply:
thanks for explaining this part!